tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,455 @@
+
+        # Directory should *still* only have one file (the trained model checkpoint).
+        assert len(list(checkpoint_dir.iterdir())) == 1
+
+    @pytest.mark.parametrize("with_grad", [True, False])
+    @pytest.mark.parametrize(
+        "device_str", ["cpu"] + (["cuda"] if torch.cuda.is_available() else [])
+    )
+    def test_configuration(
+        self,
+        profet_train_config: MetaModelConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+        with_grad: bool,
+        device_str: str,
+    ):
+        """Test that tasks have a proper configuration and that they can be created from it."""
+        max_trials = 10
+        kwargs = dict(
+            max_trials=max_trials,
+            model_config=profet_train_config,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+            device=device_str,
+            with_grad=with_grad,
+        )
+        first_task = self.Task(**kwargs)
+
+        assert first_task.configuration == {
+            self.Task.__qualname__: dict(
+                max_trials=max_trials,
+                model_config=asdict(profet_train_config),
+                input_dir=str(profet_input_dir),
+                checkpoint_dir=str(checkpoint_dir),
+                device=device_str,
+                with_grad=with_grad,
+            )
+        }
+        from orion.benchmark.benchmark_client import _get_task
+
+        # NOTE: This 'name' is the qualname of the class, which is used by the Factory metaclass'
+        # `__call__` method.
+        name, config_dict = first_task.configuration.popitem()
+        second_task = _get_task(name=name, **config_dict)
+        assert second_task.configuration == first_task.configuration
+
+    def test_space_attribute(
+        self, profet_train_config: MetaModelConfig, profet_input_dir: Path, checkpoint_dir: Path,
+    ):
+        """
+        BUG: Space.keys() returns *sorted* keys that don't match the ordering in
+        `get_search_space`, so the values returned by any space using `space.sample()` are not
+        consistent with the ordering of the returned dict from `get_search_space` of the tasks.
+
+        This wouldn't be a big deal if the samples from `space.sample()` were dictionaries, but
+        since they are tuples, there is no way of knowing what the ordering of the keys and values
+        are.
+
+        Dicts are sorted since python 3.7, there's no need to use OrderedDict anymore, and getting
+        things in a different order than you declared them is just confusing.
+
+        Doing this, for example, would be a very subtle bug:
+
+        ```
+        hparam_tuple = task.space.sample()
+        hparam = dict(zip(task.get_search_space(), hparam_tuple))
+        ```
+        This hasn't been a problem for the other tasks because they have a single dimension `x`.
+
+        This is why I'm adding a `sample` method on the task itself, so that I don't have to reach
```

Comment:
Where is the method task.sample used? Another solution would be to rely on the fact that order of dims is based on sorted keys, which is available from the task.

Reply:
> Where is the method task.sample used?

In tests.

> Another solution would be to rely on the fact that order of dims is based on sorted keys, which is available from the task.

I disagree. Doesn't make sense IMO for the space to not directly reflect the dict returned by `get_search_space`. 
IF python 3.6 support matters, then I believe `get_search_space` should be forced to return an `OrderedDict` rather than just a dict.

Reply:
Instead of adding a sample method to the public class we can add a helper function in the tests that takes the task.get_search_space() as input and return a sample as output. I think it is preferable not to add features to the public class just for the tests.

Reply:
Fixed in 679af8929ee3f8d301c6b68721201caf95e565ca